### PR TITLE
[PULL REQUEST] Updates to Hg simulation config files and other necessary fixes -- Closes #1192

### LINKED ISSUE #1192: [BUG/ISSUE] Missing oxidant fields for Hg simulation via KPP (and other issues)
### What institution are you from?
GCST

### Description of the problem
I've noticed that some of the oxidant fields that are required for the new Hg simulation via KPP have not been added to the `HEMCO_Config.rc.Hg` template file.  Also, for some of these fields we do not have archived data yet.

The issues that I've found are:

1. In routine `Set_HcoPointers`, the code expects to read the `GLOBAL_fOA` oxidant field, but we do not have that on disk as part of our 10-year benchmark output:

```console
===============================================================================
GEOS-Chem ERROR: Cannot get pointer from HEMCO for Global fOA
 -> at Set_HCOPointers (in module GeosCore/mercury_mod.F90)
===============================================================================
```

2. `Set_HcoPointers` expected to read `GLOBAL_Br` and `GLOBAL_BrO` but the Br and BrO fields are listed in HEMCO_Config.rc under different container names:

```console
#===============================================================================
# --- BrOx fields --
#==============================================================================
(((BrOx_GC
* Br_GC          $ROOT/GCClassic_Output/13.0.0/$YYYY/GEOSChem.SpeciesConc.$YYYY$MM$DD_0000z.nc4      SpeciesConc_Br    2010-2019/1-12/1/0 C xyz 1        * - 1 1
* BrO_GC         $ROOT/GCClassic_Output/13.0.0/$YYYY/GEOSChem.SpeciesConc.$YYYY$MM$DD_0000z.nc4      SpeciesConc_BrO   2010-2019/1-12/1/0 C xyz 1        * - 1 1
)))BrOx_GC

(((.not.BrOx_GC
* Br_TOMCAT      $ROOT/MERCURY/v2014-09/BrOx/BrOx.GMI.geos5.2x25.nc                                  LBRO2N                 1985/1-12/1/0 C xyz pptv     * - 1 1
* BrO_TOMCAT     $ROOT/MERCURY/v2014-09/BrOx/BrOx.GMI.geos5.2x25.nc                                  LBRO2H                 1985/1-12/1/0 C xyz pptv     * - 1 1
))).not.BrOx_GC
```

3. Entries for other oxidant fields including `GLOBAL_CO`, `GLOBAL_CH4`, `GLOBAL_PM25`, and possibly others were missing from HEMCO_Config.rc.

These issues have been causing Hg integration tests to fail.

## Solution (partial)

I was able to solve issue 2 above by adding this code into `Set_HcoPointers`:
```F90
    ! Check if the Br_Ox switch is set
    thisOpt    = HCO_GetOpt( HcoState%Config%ExtList, 'BrOx_GC', ExtNr=0 )
    is_BrOx_GC = ( INDEX( thisOpt, 'true' ) > 0 )

    ! Do for each fixed KPP species
    DO N = 1, State_Chm%nKppFix

       ! Get species ID
       spcID    = State_Chm%Map_KppFix(N)

       ! Get oxidant name
       thisName = State_Chm%SpcData(SpcID)%Info%Name

       ! Construct field name using species name
       ! Br/BrO use a different container name than the other oxidant fields
       SELECT CASE( TRIM( thisName ) )
          CASE( 'Br' )
             IF ( is_BrOx_GC ) THEN
                fieldName = 'Br_GC'
             ELSE
                fieldName = 'Br_TOMCAT'
             ENDIF

          CASE( 'BrO' )
             IF ( is_BrOx_GC ) THEN
                fieldName = 'BrO_GC'
             ELSE
                fieldName = 'BrO_TOMCAT'
             ENDIF

          CASE DEFAULT
             fieldName = 'GLOBAL_' // TRIM( thisName )

       END SELECT
```
so that will pick the proper container names for Br and BrO.  This has been fixed in commit 6d59b850dbedd51c93c60d2a81e591c8d8d18cc7.

I was also able to add the missing entries for CO, CH4, and PM25 to the  HEMCO_Config.rc.Hg template file.  This was fixed in commit 818f25e6a49bb56682e1a719faff053a0502feb9.

## Action items

1. Need data files for GLOBAL_fOA and perhaps other missing oxidant fields
2. Would appreciate it if someone could beta-test the Hg simulation and supply fixes as needed.

Tagging @jennyfisher @cdholmes @msl3v @viral211 @yanxuz @msulprizio

### PR DESCRIPTION
**Pull request by yantosca**
In this draft PR, we have updated the HEMCO_Config.rc.Hg file to read offline data for the Hg simulation.  See issue #1192 for details.

We will also add fixes for other Hg issues here as well.

### CODE REVIEW

**Comment by yantosca** [comment]
Also tagging @viral211

**Comment by yantosca** [comment]
Also tagging @cdholmes @jennyfisher @arifein @CharikleiaGRN

**Comment by yantosca** [comment]
Integration and parallelization tests pending.

**Comment by yantosca** [comment]
GCHP integration tests all passed
```console
==============================================================================
GCHP: Execution Test Results

Number of execution tests: 3
==============================================================================
 
Execution tests:
------------------------------------------------------------------------------
gchp_fullchem_benchmark_merra2_c48...............Execute Simulation.....PASS
gchp_fullchem_standard_merra2_c24................Execute Simulation.....PASS
gchp_TransportTracers_geosfp_c24.................Execute Simulation.....PASS
 
Summary of execution test results:
------------------------------------------------------------------------------
Execution tests passed:        3
Execution tests failed:        0
Execution tests not completed: 0

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%  All execution tests passed!  %%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
```

**Comment by viral211** [comment]
Thank you @yantosca. 
I am a little confused about the appropriate units for the Hg(II) and Hg(P) emissions in HEMCO_Config.rc. For example,
https://github.com/geoschem/geos-chem/blob/5f40c9917ed383aca84501cbd37a63f56186ddc8/run/GCClassic/HEMCO_Config.rc.templates/HEMCO_Config.rc.Hg#L212-L214
I had recommended using `*` instead of `kg/m2/s` for HgII and HgP emissions because the emissions are in terms of Hg mass but are assigned to species HgCl2 and Hg2ClP (different MWs than Hg). I believe if we use `kg/m2/s`  HEMCO would scale the emissions by the species' MWs, which would be wrong. What do you think we should use here, `*` or `kg/m2/s`?

File: GeosCore/mercury_mod.F90 [hunk 1]
```diff
@@ -926,72 +785,78 @@ SUBROUTINE CHEMMERCURY( Input_Opt,  State_Chm, State_Diag, &
...
-       CALL GC_Error( ErrMsg, RC, 'mercury_mod.F90' )
+       errMsg = 'Unit conversion error!'
+       CALL GC_Error( errMsg, RC, 'mercury_mod.F90' )
        RETURN
     ENDIF
 
     IF ( ITS_A_NEW_DAY() ) THEN
-        WRITE(*,*) 'Total Hg0 mass [Mg]: ',SUM ( State_Chm%Species(:,:,:,id_Hg0) )
-        WRITE(*,*) 'Total Hg2 mass [Mg]: ',SUM ( State_Chm%Species(:,:,:,2:25) )
+       WRITE(6,*) 'Total Hg0 mass [Mg]: ',                                   &
```
Comment (viral211) [inline GeosCore/mercury_mod.F90:848]: These write statements can be commented out. They were added just for testing.

File: GeosCore/mercury_mod.F90 [hunk 2]
```diff
@@ -926,72 +785,78 @@ SUBROUTINE CHEMMERCURY( Input_Opt,  State_Chm, State_Diag, &
...
-    !=======================================================================
-!    CALL PARTITIONHG2(  Input_Opt, State_Chm, State_Diag, &
-!                        State_Grid, State_Met, RC )
-
-    !=======================================================================
-    ! Hg2 uptake by seasalt aerosols in the MBL
-    !=======================================================================
-!    CALL SeaSaltUptake( Input_Opt,  State_Chm, State_Diag, &
-!                        State_Grid, State_Met, RC )
+!%% We no longer have Hg2 as a species, so these partitioning routines
```
[the remark this answers is not in the document]
Comment (yantosca) [reply GeosCore/mercury_mod.F90:821]: OK, I'll uncomment those calls again.

**Comment by yantosca** [comment]
Parallelization tests (1 core vs 8 cores) passed:
```console
Using configuration file rst18.yml
... Printing totals and differences
Variable               Ref=1 core               Dev=8 cores              Dev - Ref
AREA                 : 510065600000000.0      | 510065600000000.0      | 0.0 
Met_PS1DRY           : 3193313.8              | 3193313.8              | 0.0 
Met_PS1WET           : 3199046.2              | 3199046.2              | 0.0 
Met_DELPDRY          : 3191367.0              | 3191367.0              | 0.0 
Met_SPHU1            : 305599.25              | 305599.25              | 0.0 
Met_TMPU1            : 57776276.0             | 57776276.0             | 0.0 
SpeciesRst_Br        : 1.2751444e-07          | 1.2751444e-07          | 0.0 
SpeciesRst_BrO       : 1.3951503e-06          | 1.3951503e-06          | 0.0 
SpeciesRst_CH4       : 0.29718453             | 0.29718453             | 0.0 
SpeciesRst_CO        : 0.011681339            | 0.011681339            | 0.0 
SpeciesRst_Cl        : 3.987892e-08           | 3.987892e-08           | 0.0 
SpeciesRst_ClO       : 1.4225448e-05          | 1.4225448e-05          | 0.0 
SpeciesRst_HO2       : 3.4167497e-06          | 3.4167497e-06          | 0.0 
SpeciesRst_Hg0       : 3.8957365e-11          | 3.8957365e-11          | 0.0 
SpeciesRst_Hg2ClP    : 1.3432955e-11          | 1.3432955e-11          | 0.0 
SpeciesRst_Hg2ORGP   : 3.176606e-13           | 3.176606e-13           | 0.0 
SpeciesRst_Hg2STRP   : 2.112695e-12           | 2.112695e-12           | 0.0 
SpeciesRst_HgBr      : 4.8583856e-16          | 4.8583856e-16          | 0.0 
SpeciesRst_HgBr2     : 2.2187122e-12          | 2.2187122e-12          | 0.0 
SpeciesRst_HgBrBrO   : 1.6714938e-11          | 1.6714938e-11          | 0.0 
SpeciesRst_HgBrClO   : 9.674769e-14           | 9.674769e-14           | 0.0 
SpeciesRst_HgBrHO2   : 7.416129e-14           | 7.416129e-14           | 0.0 
SpeciesRst_HgBrNO2   : 2.710828e-13           | 2.710828e-13           | 0.0 
SpeciesRst_HgBrO     : 1.2105383e-17          | 1.2105383e-17          | 0.0 
SpeciesRst_HgBrOH    : 4.5446383e-13          | 4.5446383e-13          | 0.0 
SpeciesRst_HgCl      : 5.71136e-16            | 5.71136e-16            | 0.0 
SpeciesRst_HgCl2     : 5.604587e-13           | 5.604587e-13           | 0.0 
SpeciesRst_HgClBr    : 2.1143733e-11          | 2.1143733e-11          | 0.0 
SpeciesRst_HgClBrO   : 1.04701804e-13         | 1.04701804e-13         | 0.0 
SpeciesRst_HgClClO   : 1.11706454e-13         | 1.11706454e-13         | 0.0 
SpeciesRst_HgClHO2   : 2.3427262e-11          | 2.3427262e-11          | 0.0 
SpeciesRst_HgClNO2   : 5.2142747e-13          | 5.2142747e-13          | 0.0 
SpeciesRst_HgClO     : 3.021426e-17           | 3.021426e-17           | 0.0 
SpeciesRst_HgClOH    : 8.903027e-13           | 8.903027e-13           | 0.0 
SpeciesRst_HgOH      : 6.5909084e-16          | 6.5909084e-16          | 0.0 
SpeciesRst_HgOHBrO   : 2.0052479e-11          | 2.0052479e-11          | 0.0 
SpeciesRst_HgOHClO   : 1.3564299e-13          | 1.3564299e-13          | 0.0 
SpeciesRst_HgOHHO2   : 1.2301999e-13          | 1.2301999e-13          | 0.0 
SpeciesRst_HgOHNO2   : 2.4936027e-11          | 2.4936027e-11          | 0.0 
SpeciesRst_HgOHO     : 4.5222603e-17          | 4.5222603e-17          | 0.0 
SpeciesRst_HgOHOH    : 5.978362e-13           | 5.978362e-13           | 0.0 
SpeciesRst_NO        : 0.00021218152          | 0.00021218152          | 0.0 
SpeciesRst_NO2       : 0.0005079994           | 0.0005079994           | 0.0 
SpeciesRst_O3        : 0.4079343              | 0.4079343              | 0.0 
SpeciesRst_OH        : 4.200461e-06           | 4.200461e-06           | 0.0 
```

File: GeosCore/mercury_mod.F90 (hunk 2 above)
[the remark this answers is not in the document]
Comment (yantosca) [reply GeosCore/mercury_mod.F90:821]: > @viral211: One question, in SeaSaltUptake we have:
> 
> ```fortran-modern
>        ! Proceed only if gridbox below the stratopause
>        IF ( L > State_Met%PBL_TOP_L(I,J) ) CYCLE
> ```
> 
> PBL_TOP_L is the PBL top but the comment says that we will skip unless below the stratopause. Should we do this only in the PBL or in within the stratosphere?

I think the code is right but the comment is wrong.

**Comment by yantosca** [comment]
> @viral211: One question, in SeaSaltUptake we have:
> 
> ```fortran-modern
>        ! Proceed only if gridbox below the stratopause
>        IF ( L > State_Met%PBL_TOP_L(I,J) ) CYCLE
> ```
> 
> PBL_TOP_L is the PBL top but the comment says that we will skip unless below the stratopause. Should we do this only in the PBL or in within the stratosphere?

I think the code is right but the comment is wrong.

**Comment by viral211** [comment]
Hi @yantosca. Yes, the comment is wrong.

**Comment by yantosca** [comment]
In commit 1df8bfc726d91c02be319c47b677a800faa7a047, I have done some more general cleanup and have tried to make sure that all parallel loop variables are zeroed/initialized at the top of each loop.  But at present I have introduced a parallelization error as we do not get strict bitwise identicality between 1 and 8 cores.  I will fix this issue in a subsequent commit.

**Comment by yantosca** [comment]
Remaining parallelization issues (and more code cleanup) is now done in commit 6199ec95fa4447b5719fc9121f1344bbe19ccf67,

**Comment by yantosca** [comment]
We may have an initialization issue in routine PartitionHg2.   Still investigating.

**Comment by yantosca** [comment]
I have fixed the remaining issues in commit cdd5a5c2b3f9c124c2db81563f064e0bb449eceb.  I will run a new set of integration tests but I think this is pretty much ready to merge.  @viral211 @msulprizio, please take a look at these updates and let me know if you see any errors or have any concerns.

**Comment by yantosca** [comment]
A new parallelization tests shows identical results between 1 and 8 cores:
```console
Using configuration file rst18.yml
... Printing totals and differences
... Only showing variables with |absolute difference| > 0.0
Variable               Ref=1 core               Dev=8 cores              Dev - Ref
```

File: GeosCore/mercury_mod.F90 (hunk 2 above)
[the remark this answers is not in the document]
Comment (yantosca) [reply GeosCore/mercury_mod.F90]: We now call PartitionHg2 once again and the routine has been cleaned up/optimized.

**Comment by yantosca** [comment]
> 



> Thank you @yantosca. I am a little confused about the appropriate units for the Hg(II) and Hg(P) emissions in HEMCO_Config.rc. For example,
> 
> https://github.com/geoschem/geos-chem/blob/5f40c9917ed383aca84501cbd37a63f56186ddc8/run/GCClassic/HEMCO_Config.rc.templates/HEMCO_Config.rc.Hg#L212-L214
> 
> I had recommended using `*` instead of `kg/m2/s` for HgII and HgP emissions because the emissions are in terms of Hg mass but are assigned to species HgCl2 and Hg2ClP (different MWs than Hg). I believe if we use `kg/m2/s` HEMCO would scale the emissions by the species' MWs, which would be wrong. But I am not sure. What do you think we should use here, `*` or `kg/m2/s`?
> 
> I forgot that you had commented on this earlier: [#1063 (comment)](https://github.com/geoschem/geos-chem/pull/1063#issuecomment-995211948)

I can fix that to read all emissions with *.  I have to make another fix to the HEMCO_Config.rc file as pointed out by @msulprizio to put the readings of the oxidant fields in RUNDIR variables so that they will be available for GCAP2 met as well.

**Comment by yantosca** [comment]
@msulprizio: I have moved the oxidant field HEMCO definitions into the `gmao_hemco.txt` and `gcap2_hemco.txt` template files in commit 8dd5843f040c616dcc4b5280b70e43367d3c2c2a.  Also in this commit I implemented @viral211's suggestion so that we use `*` for all emissions units in `HEMCO_Config.rc`.

**Comment by viral211** [comment]
Hi @yantosca, Sorry I was not clear about the emission units in HEMCO_Config.rc. I was referring to the units for HgCL2 and HgCLP emissions (Hg0 emission units are fine). I am confused about how HEMCO interprets a `*` vs. `kg/m2/s` for these species. The issue is that for these species the emissions in the diagnostic output are higher than those in the input files. I think this is something to do with the fact that the emissions are in terms of Hg mass (MW=201) and we are assigning them to species with higher MWs. 

My question is this an issue with the Hg emissions diagnostic or is it that HEMCO scaling the emissions before feeding it to the model? I may have to just run a test to find out, but I wanted to see if you had an idea of what must be going on.

**Comment by viral211** [comment]
Hi @msulprizio, I do not think the specific year matters that much for the Hg oxidant fields. But I can generate Hg oxidant fields from the output using the GCAP meteorology, if that helps.

**Comment by ltmurray** [comment]
@msulprizio, the oxidants fields in $ROOT/GCAP2/OFFLINE_FIELDS/13.0.0 are the same as in $ROOT/OFFLINE_FIELDS, they have just been re-gridded to the vertical resolution of the GCAP2 meteorology.

Note, that the Hg simulation is the one speciality simulation that presently won't work with GCAP2, because I did not archive the ten extra 2-D sea-ice fraction met fields (SEAICEXX) only used by the Hg simulation for storage reasons. Users who wish to do a GCAP2 Hg simulation will have to modify the code to use the regular sea ice fraction field.

**Comment by yantosca** [comment]
> If I understand correctly, I believe `RUNDIR_Hg_Global_*` (i.e. the field used for Hg simulations) in `run/shared/settings/gcap2_hemco.txt` should remain the same as `RUNDIR_GLOBAL_*` (i.e. the field used for all other specialty simulations) in that file. For example, in `gcap2_hemco.txt`, we should have
> 
> ```
> RUNDIR_GLOBAL_Cl='* GLOBAL_Cl      $ROOT/GCAP2/OFFLINE_FIELDS/13.0.0/GEOSChem.SpeciesConc.2010-2019.$MM.$GCAP2VERTRESL.nc4 SpeciesConc_Cl         2015/1-12/1/0 C xyz 1        * - 1 1'
> ...
> ```
> 
> and
> 
> ```
> RUNDIR_Hg_GLOBAL_Cl='* GLOBAL_Cl    $ROOT/GCAP2/OFFLINE_FIELDS/13.0.0/GEOSChem.SpeciesConc.2010-2019.$MM.$GCAP2VERTRESL.nc4 SpeciesConc_Cl         2015/1-12/1/0 C xyz 1        * - 1 1
> etc.
> ```
> 
> I don't believe the GCAP2 simulation should use fields in `MERCURY/v2022-04/Oxidants_for_Hg_sim_2016.nc4`. The reason being that the fields in `GCAP2/OFFLINE_FIELDS/13.0.0/GEOSChem.SpeciesConc.2010-2019` were saved out specifically from a simulation using GCAP2 meteorology while the updated fields were saved our for 2016 only using GMAO meteorology. The ideal solution would probably be to update the `GCAP2/OFFLINE_FIELDS/` for a more recent GEOS-Chem version, but perhaps this is planned for the next major version release (14.0.0) expected later this spring or summer.
> 
> @viral211 @ltmurray Please correct me if I'm wrong.

Hi @msulprizio... the RUNDIR_Hg_GLOBAL_* fields are different because they are in molec/cm3 and not v/v dry.  I didn't want to clobber anything else so I made separate containers for the Hg inputs.

**Comment by yantosca** [comment]
> Hi @yantosca, Sorry I was not clear about the emission units in HEMCO_Config.rc. I was referring to the units for HgCL2 and HgCLP emissions (Hg0 emission units are fine). I am confused about how HEMCO interprets a `*` vs. `kg/m2/s` for these species. The issue is that for these species the emissions in the diagnostic output are higher than those in the input files. I think this is something to do with the fact that the emissions are in terms of Hg mass (MW=201) and we are assigning them to species with higher MWs.
> 
> My question is this an issue with the Hg emissions diagnostic or is it that HEMCO scaling the emissions before feeding it to the model? I may have to just run a test to find out, but I wanted to see if you had an idea of what must be going on.

Oh I see @viral211.  I can revert the Hg0 units to kg/m2/s then. 

I think if HEMCO sees a `*` then it might not do any emissions scaling, but I'm not sure. @christophkeller or @jimmielin would know for sure.

**Comment by yantosca** [comment]
> @msulprizio, the oxidants fields in $ROOT/GCAP2/OFFLINE_FIELDS/13.0.0 are the same as in $ROOT/OFFLINE_FIELDS, they have just been re-gridded to the vertical resolution of the GCAP2 meteorology. (Everything in $ROOT/GCAP2 is a 3-D field that has been vertically regridded from their 47 or 72-layer equivalents for GEOS-driven runs).
> 
> Note, that the Hg simulation is the one speciality simulation that presently won't work with GCAP2, because I did not archive the ten extra 2-D sea-ice fraction met fields (SEAICEXX) only used by the Hg simulation for storage reasons. Users who wish to do a GCAP2 Hg simulation will have to modify the code to use the regular sea ice fraction field.

Thanks @ltmurray.  We can note that the Hg simulation in GCAP is currently broken.  We would need someone to generate those oxidant fields from a long-term simulation anyway.

**Comment by ltmurray** [comment]
> > @msulprizio, the oxidants fields in $ROOT/GCAP2/OFFLINE_FIELDS/13.0.0 are the same as in $ROOT/OFFLINE_FIELDS, they have just been re-gridded to the vertical resolution of the GCAP2 meteorology. (Everything in $ROOT/GCAP2 is a 3-D field that has been vertically regridded from their 47 or 72-layer equivalents for GEOS-driven runs).
> > Note, that the Hg simulation is the one speciality simulation that presently won't work with GCAP2, because I did not archive the ten extra 2-D sea-ice fraction met fields (SEAICEXX) only used by the Hg simulation for storage reasons. Users who wish to do a GCAP2 Hg simulation will have to modify the code to use the regular sea ice fraction field.
> 
> Thanks @ltmurray. We can note that the Hg simulation in GCAP is currently broken. We would need someone to generate those oxidant fields from a long-term simulation anyway.

I can regrid the Hg oxidant fields as they exist to the GCAP2 vertical resolution if any user asks. But, she or he would have to figure out the sea ice issue first.

**Comment by yantosca** [comment]
> Hi @yantosca, Sorry I was not clear about the emission units in HEMCO_Config.rc. I was referring to the units for HgCL2 and HgCLP emissions (Hg0 emission units are fine). 

@viral211: I have restored the original units for Hg0 emissions to kg/m2/s in commit 72d8f2c9ab147619129bd6bbd0edd0317bfd2d16.

**Comment by CharikleiaGRN** [comment]
Ciao!
In the HEMCO_Config.rc.gmao_metfields, I think 
*FLASH_DENS $ROOT/OFFLINE_LIGHTNING/v2020-03/$MET/$YYYY/FLASH_CTH_$MET_0.25x0.3125*, should be replaced with 

*v2020-03/$MET/$YYYY/FLASH_CTH_$MET_0.5x0.625*

Thank you.



#==============================================================================
# --- Fields for lightning emissions (Extension 103) ---
#
# Native-resolution lightning flash rates constrained to LIS/OTD climatology
# and convective depth are now prescribed as offline meteorology.
#                                                    --Lee Murray (16 Jan 2019)
#==============================================================================
(((LightNOx
* FLASH_DENS $ROOT/OFFLINE_LIGHTNING/v2020-03/$MET/$YYYY/FLASH_CTH_$MET_0.25x0.3125_$YYYY_$MM.nc4  LDENS 1980-2021/1-12/1-31/0-23/+90minute RFY3 xy  1  * -  1 1
* CONV_DEPTH $ROOT/OFFLINE_LIGHTNING/v2020-03/$MET/$YYYY/FLASH_CTH_$MET_0.25x0.3125_$YYYY_$MM.nc4  CTH   1980-2021/1-12/1-31/0-23/+90minute RFY3 xy  1  * -  1 1
)))LightNOx

**Comment by jennyfisher** [comment]
@viral211 @yantosca - responding to the units issue.

Viral & I had a back and forth about this last year. I did a few tests and wrote the following at that point:

> My emission input files are in kg Hg/m2/s (with EmMW_g = 201 as above). As you’ve suggested, I am using HGCL2 as the species to read the emissions into.
> 
> When I output the emissions and check the output in the GEOSChem.MercuryEmis… netcdf files, the Hg0 emission values are identical to those in the inputs (once I’ve converted inputs to kg/s), confirming that I’m doing the tests correctly and that inputs are in kg Hg0/m2/s.
> 
> However, the Hg2HgP emissions are larger. If I assume the MercuryEmis output is in kg HgCl2/m2/s and multiply the result by the ratio of molecular weights (201/271.5), I recover the original value from the input files – implying the outputs are in kg HgCl2/m2/s, not kg/m2/s…
> 
> I then did a test where I changed the MW_g in the code above to 201.0_fp (to match the EmMW_g). As expected, this changes the output in the MercuryEmis file (and the HEMCO_diagnostics file), and provides the expected values that match the input emission files.

Viral responded:

> You are right, this is happening in HEMCO (in subroutine HCO_UNIT_GetMassScal). The HgCl2 emissions are being scaled by the MW_g to emMW_g ratio before they get added to the species mass. It is not just in the emissions diagnostics. 

I fixed this at the time by adding a scale factor in HEMCO_Config.rc that is equal to 201/271.5 (to effectively trick HEMCO so it multiplies by 201/271.5 here before then doing the unit conversion and multiplying by 271.5/201) and found that gave me the right emission totals.

Viral had also responded (just after I did my fix) by recommending the following:

> In the HEMCO_Config.rc file, change these two lines:
```
> 0 STREETS_HG2 $ROOT/MERCURY/v2020-07/Streets/Streets2019_Hg.nc Hg2 2000-2015/1/1/0 C xy kg/m2/s HGCL2   - 1 1
> 0 STREETS_HGP $ROOT/MERCURY/v2020-07/Streets/Streets2019_Hg.nc HgP 2000-2015/1/1/0 C xy kg/m2/s HGCL2   - 1 1

```
> To:
```
> 0 STREETS_HG2 $ROOT/MERCURY/v2020-07/Streets/Streets2019_Hg.nc Hg2 2000-2015/1/1/0 C xy * HGCL2   - 1 1
> 0 STREETS_HGP $ROOT/MERCURY/v2020-07/Streets/Streets2019_Hg.nc HgP 2000-2015/1/1/0 C xy * HGCL2   - 1 1

```
> The problem is that the units in the netcdf file are “kg m-2 s-1” but HEMCO was expecting “kg/m2/s”. Whenever HEMCO sees a mismatch in units, it does a unit conversion by scaling emissions by the ratio of the emitted MW to the species MW, which it shouldn’t in such cases. Replacing the units above to ‘*’ forces HEMCO to use the netcdf units as is.

I hope that sheds some light on which method is correct?? I've re-forwarded the email thread to Viral to check.

-Jenny

**Pull request merged**